File: cses/introdutory/permutations.py
n = int(input())

if n < 4 and n != 1:
    print("NO SOLUTION")
else:
    for x in range(1, n+1):
        if x % 2 == 0:
            print(x, end=" ")

    for x in range(1, n+1):
        if x % 2 != 0:
            print(x, end=" ")

File: cses/introdutory/permutations.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    
    int n;
    cin >> n;

    if(n < 4 && n != 1) {
        cout << "NO SOLUTION" << endl;
    } else {
        for(int i=1; i<=n; i++) {
            if(i % 2 == 0) {
                cout << i << " ";
            }
        }

        for(int i=1; i<=n; i++) {
            if(i % 2 != 0) {
                cout << i << " ";
            }
        }
    }

    return 0;
}
